superset/models/core.py

```diff
@@ -636,14 +636,24 @@ def mutate_sql_based_on_config(self, sql_: str, is_split: bool = False) -> str:
             )
         return sql_
 
-    def get_df(  # pylint: disable=too-many-locals
+    def render_sql(self, sql: str, **kwargs: Any) -> str:
```

Comment:
this method (`Database.render_sql`) allows quite a bite of simplification as the template renderer was getting passed around like crazy